app/packages/state/src/hooks/useClearSessionColorScheme.ts

```diff
@@ -19,17 +20,31 @@ const useClearSessionColorScheme = () => {
   const setSessionColorSchemeState = useSetRecoilState(sessionColorScheme);
   const dataset = useRecoilValue(datasetName);
   const stages = useRecoilValue(view);
+  const defaultSetting = useRecoilValue(datasetAppConfig).colorScheme;
 
   function onClear(saveToApp: boolean) {
+    // when clear default, we reset to app default;
+    // when reset, we reset to appConfig default (if exisits) or app default.
     const combined = {
-      colorPool: DEFAULT_APP_COLOR_SCHEME.colorPool,
-      customizedColorSettings: DEFAULT_APP_COLOR_SCHEME.customizedColorSettings,
+      colorPool:
+        defaultSetting && !saveToApp
+          ? defaultSetting.colorPool
+          : DEFAULT_APP_COLOR_SCHEME.colorPool,
+      customizedColorSettings:
+        defaultSetting?.customizedColorSettings && !saveToApp
+          ? JSON.parse(defaultSetting.customizedColorSettings)
+          : DEFAULT_APP_COLOR_SCHEME.customizedColorSettings,
     };
     const api = {
-      colorPool: DEFAULT_APP_COLOR_SCHEME.colorPool,
-      customizedColorSettings: null,
+      colorPool:
+        defaultSetting && !saveToApp
+          ? defaultSetting.colorPool
+          : DEFAULT_APP_COLOR_SCHEME.colorPool,
+      customizedColorSettings:
+        defaultSetting?.customizedColorSettings && !saveToApp
+          ? defaultSetting.customizedColorSettings
+          : null,
     };
-
     setSessionColorSchemeState(combined);
```

Comment:
Small detail, but usually I put state updates like this in the `onComplete` callback of the mutation to ensure the API call was successful and things are synced

Reply:
Yes, just added it. Thanks for catching that!